src/Optimization/hiopKKTLinSysSparse.cpp

```diff
@@ -314,32 +320,24 @@ namespace hiop
         nlp_->log->printf(verbosity,
                           "KKT_SPARSE_XYcYd linsys: alloc STRUMPACK size %d (%d cons) (safe_mode=%d)\n",
                           n, neq+nineq, safe_mode_);
-        
+
         p->setFakeInertia(neq + nineq);
         linSys_ = p;
-#else 
-        //Return NULL (and assert) if a GPU sparse linear solver is not present
-        assert(linSys_!=nullptr &&
-               "HiOp was built without a sparse linear solver for GPU/device and cannot run on the "
-               "device as instructed by the 'compute_mode' option. Change the 'compute_mode' to "
-               " 'cpu' (from hiopKKTLinSysCompressedSparseXYcYd)"); 
-        return nullptr;
-// #ifdef HIOP_USE_COINHSL
-//         nlp_->log->printf(hovScalars,
-//                           "KKT_SPARSE_XYcYd linsys: alloc MA57 on CPU size %d (%d cons)\n",
-//                           n, neq+nineq);                             
-//         linSys_ = new hiopLinSolverIndefSparseMA57(n, nnz, nlp_);
-// #endif // HIOP_USE_COINHSL
-      
-//         if(NULL == linSys_) {
-// #ifdef HIOP_USE_PARDISO
-//           nlp_->log->printf(hovScalars,
-//                             "KKT_SPARSE_XYcYd linsys: alloc PARDISO on CPU size %d (%d cons)\n",
-//                             n, neq+nineq);                             
-//           linSys_ = new hiopLinSolverIndefSparseMA57(n, nnz, nlp_);
-// #endif // HIOP_USE_PARDISO
-//        }
-#endif // HIOP_USE_STRUMPACK
+#elif  defined(HIOP_USE_COINHSL)
```

Comment:
[nitpick] Using constexpr variables from `hiopVersion.hpp` would be more readable in my opinion.

Reply:
See #417. This is imho outside the scope of this PR.